services/blockchain-connector/shared/sdk/blocksUtils.js

```diff
@@ -80,28 +76,52 @@ const loadConfig = async () => {
 	}
 };
 
-const downloadGenesisBlock = async () => {
+// eslint-disable-next-line consistent-return
+const downloadAndValidateGenesisBlock = async (retries = 2) => {
 	const directoryPath = path.dirname(genesisBlockFilePath);
-	if (!(await exists(directoryPath))) await mkdir(directoryPath, { recursive: true });
-
-	logger.info(`Downloading genesis block to the filesystem from: ${genesisBlockUrl}`);
-
-	if (genesisBlockUrl.endsWith('.tar.gz')) await downloadAndExtractTarball(genesisBlockUrl, directoryPath);
-	else await downloadJSONFile(genesisBlockUrl, genesisBlockFilePath);
-
-	// Download tar file
-	await downloadFile(genesisBlockUrl, directoryPath);
+	const genesisFileName = genesisBlockUrl.substring(genesisBlockUrl.lastIndexOf('/') + 1);
+	const genesisFilePath = `${directoryPath}/${genesisFileName}`;
+	const checksumFilePath = `${genesisFilePath}.SHA256`;
+
+	do {
+		/* eslint-disable no-await-in-loop */
+		try {
+			if (!(await exists(directoryPath))) await mkdir(directoryPath, { recursive: true });
+
+			// Download the genesis and the digest files
+			const genesisBlockUrlSHA256 = genesisBlockUrl.concat('.SHA256');
+			await downloadFile(genesisBlockUrl, directoryPath);
+			await downloadFile(genesisBlockUrlSHA256, directoryPath);
+
+			// Verify the integrity of the downloaded file, retry on failure
+			const isValidGenesisBlock = await verifyFileChecksum(genesisFilePath, checksumFilePath);
+
+			// Extract the file if necessary
+			if (genesisFilePath.endsWith('.tar.gz')) {
+				await extractTarBall(genesisFilePath, directoryPath);
+			}
+
+			if (isValidGenesisBlock) return true;
```

Comment:
Please extract the tarball only when the integrity check is successful.
```suggestion
			if (isValidGenesisBlock) {
				// Extract the file if necessary
				if (genesisFilePath.endsWith('.tar.gz')) {
					await extractTarBall(genesisFilePath, directoryPath);
				}

				return true;
			}
```

Reply:
Applied